core/src/main/java/com/bankbazaar/core/manager/TripActivityEntityManager.java

```diff
@@ -0,0 +1,97 @@
+package com.bankbazaar.core.manager;
+
+import com.bankbazaar.core.model.TripActivityCompositeKey;
+import com.bankbazaar.core.model.TripActivityEntity;
+import com.bankbazaar.core.model.TripEntity;
+import com.bankbazaar.core.repository.TripActivityRepository;
+import com.bankbazaar.core.repository.TripRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+@Slf4j
+@Service
+public class TripActivityEntityManager {
+    @Autowired
+    private TripActivityRepository tripActivityRepository;
+
+    @Autowired
+    private TripRepository tripRepository;
+
+    /**
+     * Insert to TripActivity table
+     * @param data
+     */
+    public TripActivityEntity saveTripActivity(TripActivityEntity data)
+    {
+        Optional<TripEntity> tripId = tripRepository.findById(data.getTripId().getTripId());
+        data.setTripId(tripId.get());
+        return tripActivityRepository.save(data);
+    }
+
+    /**
+     * update to TripActivity table
+     * @param data
+     */
+    public TripActivityEntity updateTripActivity(TripActivityEntity data)
+    {
+        Optional<TripActivityEntity> presentData = exists(data.getTripId().getTripId(),data.getActivityId());
```

Comment:
If you have the activity id already do we need to fetch using tripId?

Reply:
This was unnecessary. I had implemented this right after TripUserMapping, so i followed the same structure. I have made changes to this.